lib/rule-list.js

```diff
@@ -423,13 +423,20 @@ export const MULTIPLY_COEFFICIENTS = defineRule(
                 }
             }
         })
-        node.args.forEach(arg =>
-                          coeffs.push(getCoefficient(arg)))
-        const result = build.apply(
-            'mul',
-            [build.mul(...coeffs), build.mul(...terms)],
-            {implicit: true}
-        )
+        node.args.forEach(function(arg) {
+            if (query.getValue(getCoefficient(arg)) != 1) {
+                coeffs.push(getCoefficient(arg))
+            }
+        })
+
+        let poly = build.mul(...terms)
+        
+        const result =  coeffs.length > 1
+              ? build.implicitMul(build.mul(...coeffs), poly)
+              : coeffs.length == 1
+              ? build.implicitMul(coeffs[0], poly)
+              : poly
```

Comment:
Please no nested ternaries.  They're hard to understand and even harder understand with this formatting.